### PR TITLE
alphanet: merge build changes and consensus params

### PR DESCRIPTION
**Pull request by cce**
## Summary

This merges in a few changes related to alphanet
- build changes to add an alphanet channel (alongside betanet, devnet, testnet, etc)
- alphanet genesis file
- alphanet test scenario recipe
- alphanet consensus parameters

## Test Plan

Existing tests should pass.
Added a new test to assert that non-official consensus version names (like "alpha3") don't make it into the upgrade history to ConsensusCurrentVersion.

### CODE REVIEW

**Review by onetechnical — APPROVED** [review]
The topology seems a little strange for me, but otherwise this looks good

**Review by algolucky — DISMISSED** [review]

**Review by egieseke — DISMISSED** [review]
Looks good.

**Review by algolucky — DISMISSED** [review]

**Review by egieseke — APPROVED** [review]

**Review by algolucky — APPROVED** [review]

**Review by egieseke — APPROVED** [review]

File: test/testdata/deployednettemplates/recipes/alphanet/genesis.json [hunk 1]
```diff
@@ -1,59 +1,159 @@
 {
-	"NetworkName": "alphanet",
+	"NetworkName": "",
 	"VersionModifier": "",
-	"ConsensusProtocol": "alpha1",
```
Comment (algorandskiy) [inline test/testdata/deployednettemplates/recipes/alphanet/genesis.json:4]: why did we change genesis'proto?

File: config/config_test.go [hunk 2]
```diff
@@ -345,21 +346,35 @@ func TestConsensusUpgrades(t *testing.T) {
...
 	latestVersionName := protocol.ConsensusCurrentVersion
 
-	leadsTo := consensusUpgradesTo(a, currentVersionName, latestVersionName)
+	leadsTo := consensusUpgradesTo(a, currentVersionName, latestVersionName, checkConsensusVersionName)
 	a.True(leadsTo, "Consensus protocol must have upgrade path from %v to %v", currentVersionName, latestVersionName)
 }
 
-func consensusUpgradesTo(a *require.Assertions, currentName, targetName protocol.ConsensusVersion) bool {
+func checkConsensusVersionName(a *require.Assertions, name string) {
+	// ensure versions come from official specs repo
```
Comment (algorandskiy) [inline config/config_test.go:354]: why is this check important?

**Review by algorandskiy — COMMENTED** [review]
I do not know much about alphanet but LGTM

File: test/testdata/deployednettemplates/recipes/alphanet/genesis.json (hunk 1 above)
Comment (onetechnical) [reply test/testdata/deployednettemplates/recipes/alphanet/genesis.json:4]: I believe we update this for performance testing. The live network genesis doesn't change (it's in the installer dir) but we need to keep this updated.

File: config/config_test.go (hunk 2 above)
Comment (onetechnical) [reply config/config_test.go:354]: I believe this is a sanity check to ensure we don't accidentally enable vAlpha in official builds.

Comment (cce) [reply config/config_test.go:354]: yes it was an idea I had before merging these alternate consensus versions, that I should add something to be extra careful to ensure a version like "alpha3" doesn't make it into the path from v7 to ConsensusCurrentVersion and into a release

File: test/testdata/deployednettemplates/recipes/alphanet/genesis.json (hunk 1 above)
Comment (cce) [reply test/testdata/deployednettemplates/recipes/alphanet/genesis.json:4]: This is just to make the Jenkins tests running the "alphanet" recipe start up from the latest version — the real alphanet genesis is in installer/genesis/alphanet/genesis.json. I guess another approach would be to have an `alphalatest` or `alphafuture` version defined to set this to

**Pull request merged**